zeus/tasks/aggregate_job_stats.py

```diff
@@ -149,11 +149,18 @@ def aggregate_build_stats(build_id: UUID):
         build.date_finished = None
 
     # if theres any failure, the build failed
-    if any(j.result is Result.failed for j in job_list):
+    if any(j.result is Result.failed for j in job_list if not j.allow_failure):
         build.result = Result.failed
     # else, if we're finished, we can aggregate from results
     elif is_finished:
-        build.result = aggregate_result((j.result for j in job_list))
+        build.result = aggregate_result((j.result for j in job_list if not j.allow_failure))
```

Comment:
fwiw, the second pair of parentheses is unnecessary. `aggregate_result(j.result for j in job_list if not j.allow_failure)` works fine.

Reply:
i feel like historically things have yelled at me about not enclosing generator expressions

ill do a pass to clean them up in a future pr